terminology/src/main/java/au/csiro/pathling/config/HttpClientCachingConfiguration.java

```diff
@@ -0,0 +1,170 @@
+import java.io.Serializable;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.validation.Constraint;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.Payload;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.apache.http.impl.client.cache.CacheConfig;
+
+/**
+ * Represents configuration relating to HTTP client caching.
+ */
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@ValidHttpCacheConfiguration
+public class HttpClientCachingConfiguration implements Serializable {
+
+  private static final long serialVersionUID = -3030386957343963899L;
+
+  public static final boolean DEFAULT_ENABLED = true;
+  public static final int DEFAULT_MAX_CACHE_ENTRIES = 100_000;
+  public static final long DEFAULT_MAX_OBJECT_SIZE = 64_000L;
+  public static final StorageType DEFAULT_STORAGE_TYPE = StorageType.MEMORY;
+
+  /**
+   * Enables client side caching of REST requests.
+   */
+  @NotNull
+  @Builder.Default
+  private boolean enabled = DEFAULT_ENABLED;
+
+  /**
+   * Sets the maximum number of entries the cache will retain.
+   * <p>
+   * See also: {@link CacheConfig.Builder#setMaxCacheEntries(int)}
+   */
+  @NotNull
+  @Min(0)
+  @Builder.Default
+  private int maxEntries = DEFAULT_MAX_CACHE_ENTRIES;
+
+  /**
+   * Sets the maximum size of a cacheable response, in bytes.
+   * <p>
+   * See also: {@link CacheConfig.Builder#setMaxObjectSize(long)}
+   */
+  @Min(0)
+  @NotNull
+  @Builder.Default
+  private long maxObjectSize = DEFAULT_MAX_OBJECT_SIZE;
+
+  @NotNull
+  @Builder.Default
+  private StorageType storageType = DEFAULT_STORAGE_TYPE;
+
+  @Nullable
+  private String storagePath;
+
+  /**
+   * Represents the type of storage used by the cache.
+   */
+  public enum StorageType {
+    /**
+     * The cache is stored in memory, and is reset when the application is restarted.
+     */
+    MEMORY("memory"),
+
+    /**
+     * The cache is stored on disk, and is persisted between application restarts.
+     */
+    DISK("disk");
+
+    @Nonnull
+    private final String code;
+
+    StorageType(@Nonnull final String code) {
+      this.code = code;
+    }
+
+    @Override
+    public String toString() {
+      return code;
+    }
+
+    @Nullable
+    public static StorageType fromCode(@Nullable final String code) {
+      for (final StorageType storageType : values()) {
+        if (storageType.code.equals(code)) {
+          return storageType;
+        }
+      }
+      return null;
+    }
+  }
+
```

Comment:
Maybe make @Nonnull and throw illegal argument exception here;

Reply:
Ok, I think I understand what you are saying now. I've made the change in b3d8ed0044023ce196eadfdf837681e5e3311d90.